### ISSUE TITLE
Some issues about the morlhf/ppo code

### ISSUE THREAD

**Issue by andyclsr** [YangRui2015/RiC#10]
Hi, sorry for bothering you again. I have some issues with the generation config of ppo.
As shown below, `pad_token_id` and `begin_suppress_tokens` are set to be eos token. I wonder are there any explanations for this setting? I found that it is rare to have `begin_suppress_tokens` set. What is the reason for this?
https://github.com/YangRui2015/RiC/blob/f8fa613d1dbe28c59b83801666558f4282925dd7/ppo/ppo.py#L151-L152
And it seems like the pad token has been set in `load_main_tokenizer`. Then why still need to set pad_token as eos_token?
https://github.com/YangRui2015/RiC/blob/f8fa613d1dbe28c59b83801666558f4282925dd7/ppo/ppo.py#L109

Anticipating for your reply! Thanks anyway for your patient replies all along!
Finally, Happy Mid Autumn Festival!

**Comment by YangRui2015** [comment]
We set `begin_suppress_tokens` to the `eos_token` because we observed that PPO tends to quickly learn to generate nothing, causing errors in the pipeline. By using `begin_suppress_tokens`, we mitigate this issue by forcing the model to generate output. However, this is not an optimal solution. A better approach would involve learning a better reward model for policy learning and tuning the PPO hyperparameters, such as KL regularization and learning rate.

The `pad_token` is set for different classes (tokenizer or generator), though most of our generation tasks use a batch size of 1, meaning the `pad_token` is rarely used. We include it as a precaution.
